src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7327,21 +7357,36 @@ subroutine summarize_timestep(domain)
                              realArgs=(/global_scalar_min, latMax_global, lonMax_global/))
 
          scalar_max = -1.0e20
+         offset_1d = huge(1)
          indexMax = -1
          kMax = -1
          latMax = 0.0
          lonMax = 0.0
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(max:scalar_max)
+         do iCell = 1, nCellsSolve
+         do k = 1, nVertLevels
+            scalar_max = max(scalar_max, w(k,iCell))
+         end do
+         end do
+         !$acc end parallel
+
+         !$acc parallel default(present)
+         !$acc loop collapse(2) gang vector reduction(min:lin_indexMax)
          do iCell = 1, nCellsSolve
          do k = 1, nVertLevels
-            if (w(k,iCell) > scalar_max) then
-               scalar_max = w(k,iCell)
-               indexMax = iCell
-               kMax = k
-               latMax = latCell(iCell)
-               lonMax = lonCell(iCell)
+            if (w(k,iCell) == scalar_max) then
+               offset_1d = min(offset_1d, k + size(w,1)*(iCell-1))
             end if
          end do
          end do
+         !$acc end parallel
+         kMax = modulo(offset_1d, size(w,1))
+         indexMax = ((offset_1d - kMax) / size(w,1)) + 1
+         kMax = modulo(lin_indexMax, size(w,1))
+         indexMax = ((lin_indexMax - kMax) / size(w,1)) + 1
```

Comment:
It looks like these two lines (using `lin_indexMax`) should be replaced by the two lines above them (using `offset_1d`).